apollo-federation/src/merge.rs

```diff
@@ -1185,6 +1198,12 @@ fn join_field_applied_directive(
             value: external.into(),
         }));
     }
+    if inaccessible {
+        join_field_directive.arguments.push(Node::new(Argument {
```

Comment:
btw, there is a shorthand syntax for creating a `Node<Argument>` from a `(Name, impl Into<Value>)` tuple:
```
        join_field_directive.arguments.push((name!("inaccessible"), inaccessible).into())
```